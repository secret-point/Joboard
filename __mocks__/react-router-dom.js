// Mock navigate and useParams when needed
// const navigate = jest.fn();
// const useParams = jest.fn();
const useLocation = jest.fn();

module.exports = {
  ...jest.requireActual("react-router-dom"),
  // useNavigate: () => navigate,
  // useParams,
  useLocation,
};
