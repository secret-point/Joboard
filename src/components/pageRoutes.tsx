export enum PAGE_ROUTES {
  DEFAULT = "/",
  CONSENT = "consent",
  PRE_CONSENT = "pre-consent",
  JOB_OPPORTUNITIES = "job-opportunities",
  JOB_CONFIRMATION = "job-opportunities/job-confirmation",
  JOB_DESCRIPTION = "job-opportunities/job-description",
  CONTINGENT_OFFER = "contingent-offer",
  BACKGROUND_CHECK = "bgc",
  BACKGROUND_CHECK_FCRA = "bgc/fcra",
  RESUME_APPLICATION = "resume-application",
  NHE = "nhe",
  ALREADY_APPLIED = "already-applied",
  SELF_IDENTIFICATION = "self-identification",
  REVIEW_SUBMIT = "review-submit",
  THANK_YOU = "thank-you",
  SESSION_TIMEOUT = "session-timeout",
  WOTC = "wotc",
  WOTC_COMPLETE = "wotc-complete",
  ASSESSMENT = "assessment",
  ASSESSMENT_CONSENT = "assessment-consent",
  ASSESSMENT_NOT_ELIGIBLE = "assessment-not-eligible",
  ASSESSMENT_FINISHED = "assessment-finished",
  CANDIDATE_WITHDRAWS = "candidate-withdraws",
  CALI_DISCLOSURE = "cali-disclosure",
  AMAZON_REJECTS = "amazon-rejects",
  WORKFLOW_FAILED = "workflow-failed",
  AMAZON_WITHDRAWS = "amazon-withdraws",
  NO_AVAILABLE_TIME_SLOTS = "no-available-time-slots",
  CAN_NOT_OFFER_JOB = "can-not-offer-job",
  SUPPLEMENTARY_SUCCESS = "supplementary-success",
  REHIRE_ELIGIBILITY_STATUS = "rehire-eligibility-status",
  NO_AVAILABLE_SHIFT = "no-available-shift",
  APPLICATIONID_NULL = "applicationId-null",
  SHIFT_PREFERENCE = "job-opportunities/shift-preferences",
  ADDITIONAL_INFORMATION = "additional-information"
}

// Reference: https://code.amazon.com/packages/HVHStepFunctionDefinitions/blobs/ad75b9869a017bc84a9e03920e835a6b2cec7063/--/configuration/nacf_tier_1_node_workflow/defintion.json#L186
export const PagesControlledByWorkFlowService = [
  PAGE_ROUTES.JOB_OPPORTUNITIES,
  PAGE_ROUTES.ASSESSMENT_CONSENT,
  PAGE_ROUTES.ASSESSMENT,
  PAGE_ROUTES.CONTINGENT_OFFER,
  PAGE_ROUTES.BACKGROUND_CHECK,
  PAGE_ROUTES.CAN_NOT_OFFER_JOB,
  PAGE_ROUTES.CALI_DISCLOSURE,
  PAGE_ROUTES.NHE,
  PAGE_ROUTES.SELF_IDENTIFICATION,
  PAGE_ROUTES.REVIEW_SUBMIT,
  PAGE_ROUTES.THANK_YOU,
  PAGE_ROUTES.WOTC,
  PAGE_ROUTES.SUPPLEMENTARY_SUCCESS,
  PAGE_ROUTES.ASSESSMENT_NOT_ELIGIBLE,
  PAGE_ROUTES.REHIRE_ELIGIBILITY_STATUS,
  PAGE_ROUTES.AMAZON_WITHDRAWS,
  PAGE_ROUTES.CANDIDATE_WITHDRAWS,
  PAGE_ROUTES.AMAZON_REJECTS,
  PAGE_ROUTES.WORKFLOW_FAILED,
  PAGE_ROUTES.RESUME_APPLICATION,
  PAGE_ROUTES.SHIFT_PREFERENCE,
  PAGE_ROUTES.ADDITIONAL_INFORMATION
];
