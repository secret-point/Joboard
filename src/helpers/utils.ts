import isEmpty from "lodash/isEmpty";

export const convertPramsToJson = (params: string) => {
  if (!isEmpty(params)) {
    return JSON.parse(
      '{"' +
        decodeURI(params.substring(1))
          .replace(/"/g, '\\"')
          .replace(/&/g, '","')
          .replace(/=/g, '":"') +
        '"}'
    );
  } else {
    return {};
  }
};

export const launchAuthentication = () => {
  let hash = window.location.hash.substr(1);

  const origin = window.location.origin;
  const redirectUrl = `${origin}/#${hash}`;

  const state = window.reduxStore.getState();
  let url = `${
    state.app.appConfig.authenticationURL
  }/?redirectUrl=${encodeURIComponent(redirectUrl)}`;
  window.location.assign(url);
};

export const isJson = (obj: string) => {
  try {
    JSON.parse(obj);
  } catch (e) {
    return false;
  }
  return true;
};
