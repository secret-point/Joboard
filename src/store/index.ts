import { createStore, applyMiddleware, compose  } from "redux";
import thunk from "redux-thunk";
import rootReducer from "../reducers";
import { createHashHistory } from "history";
import { routerMiddleware  } from "react-router-redux";
import { createLogger } from "redux-logger";

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const middleware = [thunk, routerMiddleware(createHashHistory()), createLogger()];
const store = createStore(rootReducer, {} , composeEnhancers(applyMiddleware(...middleware)));
export default store;