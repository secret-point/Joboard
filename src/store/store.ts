import { applyMiddleware, createStore } from "redux";
import { routerMiddleware } from "connected-react-router";
import { createHashHistory } from "history";
import { createEpicMiddleware } from "redux-observable";
import createRootReducer from "../reducers/root.reducer";
import rootEpic from "../epics/root.epic";
import { composeWithDevTools } from "redux-devtools-extension";
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage'

/* TODO remove any from functions */
const logger = () => (next: any) => (action: any) => {
  // eslint-disable-next-line no-console
  // console.log("dispatching", action);
  const result = next(action);
  // console.log("next state", store.getState());
  // eslint-disable-next-line no-console
  return result;
};

export const history = createHashHistory();

const persistConfig = {
  key: 'root',
  storage, // defaults to localStorage for web
  whitelist: ['request']
};

export const configureStore = (preloadedState: any) => {
  const epicMiddleware = createEpicMiddleware();
  const middlewares: any = [epicMiddleware, routerMiddleware(history), logger];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer];
  const composedEnhancers: any = composeWithDevTools(...enhancers);
  const persistedReducer = persistReducer(persistConfig, createRootReducer(history));

  const store = createStore(
    persistedReducer,
    preloadedState,
    composedEnhancers
  );

  epicMiddleware.run(rootEpic);

  return store;
};

const store = configureStore({});

export const persistor = persistStore(store);

export default store;
