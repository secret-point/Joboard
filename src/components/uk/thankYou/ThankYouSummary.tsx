import React, { useEffect } from "react";
import { ButtonVariant } from "@amzn/stencil-react-components/button";
import { Card } from "@amzn/stencil-react-components/card";
import { FlyoutContent, RenderFlyoutFunctionParams, WithFlyout } from "@amzn/stencil-react-components/flyout";
import { IconArrowRight, IconSize } from "@amzn/stencil-react-components/icons";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import InnerHTML from "dangerously-set-html-content";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import { boundGetCandidateInfo } from "../../../actions/CandidateActions/boundCandidateActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  redirectToDashboard,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { ApplicationState } from "../../../reducers/application.reducer";
import { CandidateState } from "../../../reducers/candidate.reducer";
import { JobState } from "../../../reducers/job.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { CommonColors } from "../../../utils/colors";
import {
  checkAndBoundGetApplication,
  formatDate,
  getLocale,
  initiateScheduleDetailOnPageLoad
} from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import DebouncedButton from "../../common/DebouncedButton";
import { PAGE_ROUTES } from "../../pageRoutes";
import NhePreferenceReview from "../../common/nhe/NhePreferenceReview";

interface MapStateToProps {
  application: ApplicationState;
  candidate: CandidateState;
  schedule: ScheduleState;
  job: JobState;
}

export const ThankYouSummary = (props: MapStateToProps) => {
  const { application, candidate, schedule, job } = props;
  const { search, pathname } = useLocation();
  const pageName = getPageNameFromPath(pathname);
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { applicationId, jobId, scheduleId } = queryParams;
  const applicationData = application.results;
  const { scheduleDetail } = schedule.results;
  const { candidateData, shiftPreferencesData: candidateShiftPreferences } = candidate.results;
  const jobDetail = job.results;
  const nheAppointment = applicationData?.nheAppointment;
  const location = applicationData?.nheAppointment?.location;
  const startTime = nheAppointment?.startTime;
  const endTime = nheAppointment?.endTime;

  useEffect(() => {
    // Refresh and add scheduleId in the url from the jobSelected if it doesn't exist from the query param
    if (!scheduleId && applicationData) {
      initiateScheduleDetailOnPageLoad(applicationData, PAGE_ROUTES.THANK_YOU);
    }
  }, [applicationData]);

  useEffect(() => {
    boundGetCandidateInfo();
  }, []);

  useEffect(() => {
    checkAndBoundGetApplication(applicationId);
  }, [applicationId]);

  useEffect(() => {
    scheduleId && scheduleId!== scheduleDetail?.scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    });
  }, [scheduleDetail, scheduleId]);

  useEffect(() => {
    if (jobId) {
      jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
    }
  }, [jobDetail, jobId]);

  useEffect(() => {
    jobDetail && applicationData && candidateData && scheduleDetail && addMetricForPageLoad(pageName);

  }, [jobDetail, applicationData, candidateData, scheduleDetail, pageName]);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, [pageName]);

  const handleGoToDashboard = () => {
    redirectToDashboard();
  };

  const renderNheAppointmentDetails = () => {
    return (
      <>
        <Text fontSize="T300" fontWeight="bold">
          {t("BB-Kondo-ThankYou-nhe-appointment-details-title-text", "Pre-hire appointment details")}
        </Text>

        <Text fontSize="T100">
          {t("BB-Kondo-ThankYou-nhe-appointment-details-date-label", "Date")}: {formatDate(nheAppointment?.dateWithoutFormat, {
            defaultDateFormat: "DD/MM/yyyy",
            displayFormat: "dddd, D MMM"
          })}
        </Text>
        <Text fontSize="T100">
          {t("BB-Kondo-ThankYou-nhe-appointment-details-time-label", "Time")}: {`${startTime} - ${endTime}`}
        </Text>
        <Text fontSize="T100">
          {t("BB-Kondo-ThankYou-nhe-appointment-details-meeting-details-label", "Meeting details")}: {location && `${location.streetAddress} ${location.city} ${location.postalCode}, United Kingdom`}
        </Text>
      </>
    );
  };

  const renderFlyout = ({ close }: RenderFlyoutFunctionParams) => (
    <FlyoutContent
      titleText={t("BB-ThankYou-nhe-appointment-flyout-title-text", "Pre-hire appointment")}
      onCloseButtonClick={close}
    >
      <Col width="100%" padding="S300" gridGap={15}>
        {renderNheAppointmentDetails()}

        <Text fontSize="T300" fontWeight="bold">
          {t("BB-ThankYou-nhe-appointment-flyout-things-to-bring-text", "Things to bring")}
        </Text>
        <ul className="ul-list">
          <li>
            <Text fontSize="T100">
              {t("BB-ThankYou-nhe-appointment-flyout-original-unexpired-photoid-text", "Original unexpired photo ID")}
            </Text>
          </li>
          <li>
            <Text fontSize="T100">
              {t("BB-ThankYou-nhe-appointment-flyout-original-unexpired-work-authorization-document-text", "Original unexpired work authorization document")}
            </Text>
          </li>
        </ul>

        <Text fontSize="T100">
          <InnerHTML html={t("BB-Kondo-ThankYou-nhe-appointment-flyout-acceptable-documents-text", "Click <a href='https://glspub.s3-us-west-2.amazonaws.com/Work+Authorization/UK+Employee+Guide/index.html' target='_blank' rel='noopener noreferrer'>here</a> for a complete list of acceptable documents.")} />
        </Text>

        <Text fontSize="T300" fontWeight="bold">
          {t("BB-ThankYou-nhe-appointment-flyout-special-assistance-text", "Special assistance")}
        </Text>
        <Text fontSize="T100">
          <InnerHTML html={t("BB-Kondo-ThankYou-nhe-appointment-flyout-accommodations-assistance-text", "If you have a disability and consider that you may require an adjustment to Amazon's recruitment process, please contact us to discuss on 0808 164 9427 or by email at myapplication@amazon.co.uk")} />
        </Text>
      </Col>
    </FlyoutContent>
  );

  // temporarily look for either application OR candidate shift preferences at the thank-you summary so candidates
  // who may have submitted with application shift preferences but loaded when candidate shift preferences is toggled
  // on still get the correct experience
  const hasShiftPreferences = applicationData?.shiftPreference ?? candidateShiftPreferences;
  return (
    <Col gridGap="S300" padding={{ top: "S300" }}>

      {/* when shift preference selected and no shift selected, https://sim.amazon.com/issues/Kondo_QA_Issue-51 */}
      { !applicationData?.jobScheduleSelected.scheduleId && hasShiftPreferences && (
        <Col gridGap="S300">
          <Text fontSize="T400">
            {t("BB-Kondo-ThankYou-with-shift-preference-title-text", "Thank you for submitting your shift preferences!")}
          </Text>
          <Text>
            {t("BB-Kondo-ThankYou-with-shift-preference-content-text", "We'll contact you as soon as we find a shift that works for you with your chosen preferences.")}
          </Text>
        </Col>
      )}

      {/* when shift is selected and nhePreference selected */}
      { scheduleDetail && applicationData?.nhePreference && !applicationData?.nheAppointment && (
        <Col gridGap="S300">
          <Text fontSize="T400">
            {t("BB-Kondo-ThankYou-with-nhe-preference-title-text", "Thank you for submitting your pre-hire appointment preferences!")}
          </Text>
          <Text fontSize="T200">
            {t("BB-Kondo-ThankYou-with-nhe-preference-content-text", "We'll contact you as soon as we find a pre-hire appointment that works for you with your chosen preferences.")}
          </Text>
          <NhePreferenceReview nhePreference={applicationData?.nhePreference} />
        </Col>
      )}

      {/* when shift is selected and nhe appointment is selected */}
      { scheduleDetail && applicationData?.nheAppointment && (
        <Col gridGap="S300">
          <Text fontSize="T400">
            {t("BB-ThankYou-thank-you-title-text", "Thank you for applying to Amazon!")}
          </Text>
          <Text fontSize="T200">
            {t("BB-Kondo-ThankYou-description-text", "You will receive an email from us providing further information on the next steps in your application. We look forward to seeing you at your pre-hire appointment.")}
          </Text>

          <Card width="100%" padding="S300" isElevated>
            <Col width="100%" padding="S300" gridGap={15}>
              {renderNheAppointmentDetails()}

              <Row className="border-top-row">
                <WithFlyout renderFlyout={renderFlyout}>
                  {({ open }) => (
                    <Col alignItems="flex-start">
                      <Text
                        color={CommonColors.Blue70}
                        style={{ cursor: "pointer" }}
                        fontWeight="bold"
                        onClick={open}
                      >
                        <Row
                          gridGap={8}
                          alignItems="center"
                        >
                          {t("BB-Kondo-ThankYou-more-details-text", "More details")}
                          <IconArrowRight size={IconSize.ExtraSmall} aria-hidden />
                        </Row>
                      </Text>
                    </Col>
                  )}
                </WithFlyout>
              </Row>
            </Col>
          </Card>
        </Col>
      )}

      <DebouncedButton variant={ButtonVariant.Primary} onClick={handleGoToDashboard}>
        {t("BB-Kondo-ThankYou-return-to-dashboard-text", "Return to dashboard")}
      </DebouncedButton>
    </Col>
  );
};

export const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(ThankYouSummary);
