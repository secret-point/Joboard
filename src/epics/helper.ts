import { log, LoggerType } from "../helpers/log-helper";
import { ProxyApiError } from "../utils/api/types";

export const epicSwitchMapHelper = async (response: any) => {
  if (!response.data || !!response.errorCode || !!response.error || response.errorMessage) {
    log("[Epic] epicSwitchMapHelper: Starting to raise proxy api error", response, LoggerType.ERROR);
    const { errorCode } = response;
    throw createProxyApiEpicError(errorCode, response.error || response.errorMessage);
  }
  return response;
};

export const createProxyApiEpicError = (errorCode: string, errorMessage?: string): ProxyApiError => {
  return { errorCode, errorMessage };
};
