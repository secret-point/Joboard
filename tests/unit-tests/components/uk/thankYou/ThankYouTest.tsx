import { shallow } from "enzyme";
import React from "react";
import { mountWithStencil } from "@amzn/stencil-react-components/tests";
import { PAGE_ROUTES } from "../../../../../src/components/pageRoutes";
import {getFeatureFlagValue} from "../../../../../src/utils/helper";
import {
    TEST_APPLICATION_ID,
    TEST_APPLICATION_STATE,
    TEST_CANDIDATE_STATE, TEST_JOB,
    TEST_JOB_ID,
    TEST_JOB_STATE, TEST_SCHEDULE_ID
} from "../../../../test-utils/test-data";
import store from "../../../../../src/store/store";
import { Provider } from "react-redux";
import { useLocation } from "react-router-dom";
import ThankYouWithStore,{ThankYou} from "../../../../../src/components/uk/thankYou/ThankYou";
import {ThankYouSummary} from "../../../../../src/components/uk/thankYou/ThankYouSummary";
import {ThankYouRedirectToAsh} from "../../../../../src/components/uk/thankYou/ThankYouRedirectToAsh";

import Mock = jest.Mock;

jest.mock('../../../../../src/utils/helper',()=>{
    const actual = jest.requireActual('../../../../../src/utils/helper');
    return {
        ...actual,
        getFeatureFlagValue:jest.fn(),
    }
});

describe("ThankYouWrapper Component", () => {
    let getFeatureFlagValueMock = getFeatureFlagValue as Mock;
    const initStoreState = store.getState();
    let state: any;
    store.getState = () => state;

    const mockLocation = {
        pathname: "/thank-you",
        search: `?jobId=${TEST_JOB_ID}&applicationId=${TEST_APPLICATION_ID}&scheduleId=${TEST_SCHEDULE_ID}`,
        hash: "",
        state: null
    };
    const mockUseLocation = useLocation as jest.Mock;
    mockUseLocation.mockReturnValue(mockLocation);

    beforeEach(() => {
        state = {
            ...initStoreState,
            application: TEST_APPLICATION_STATE,
            job: TEST_JOB_STATE,
            candidate: TEST_CANDIDATE_STATE
        };
        mockUseLocation.mockReturnValue(mockLocation);
        getFeatureFlagValueMock.mockReset();
    });

    it("should match snapshot", () => {
        const shallowWrapper = shallow(<ThankYou job={TEST_JOB_STATE}/>);
        expect(shallowWrapper).toMatchSnapshot();
    });

    it("should render the thank you summary component when enable thank you redirect is false", () => {
        getFeatureFlagValueMock.mockReturnValueOnce(false);
        window.location.hash = `#/${PAGE_ROUTES.THANK_YOU}`;
        state = {
            ...initStoreState,
            application: TEST_APPLICATION_STATE,
            job: {
                ...TEST_JOB_STATE,
                results:{
                    ...TEST_JOB,
                    dspEnabled:true,
                }
            },
            candidate: TEST_CANDIDATE_STATE
        };
        const wrapper = mountWithStencil(
            <Provider store={store}>
                <ThankYouWithStore />
            </Provider>);

        expect(wrapper.find(ThankYouSummary)).toHaveLength(1);
        expect(getFeatureFlagValueMock).toHaveBeenCalledWith("ENABLE_THANK_YOU_REDIRECT_TO_ASH")
    });

    it("should render the thank you summary component when enable thank you redirect is true but the job is not dsp enabled", () => {
        getFeatureFlagValueMock.mockReturnValueOnce(false);
        window.location.hash = `#/${PAGE_ROUTES.THANK_YOU}`;
        state = {
            ...initStoreState,
            application: TEST_APPLICATION_STATE,
            job: {
                ...TEST_JOB_STATE,
                results:{
                    ...TEST_JOB,
                    dspEnabled:false,
                }
            },
            candidate: TEST_CANDIDATE_STATE
        };
        const wrapper = mountWithStencil(
            <Provider store={store}>
                <ThankYouWithStore />
            </Provider>);

        expect(wrapper.find(ThankYouSummary)).toHaveLength(1);
        expect(getFeatureFlagValueMock).toHaveBeenCalledWith("ENABLE_THANK_YOU_REDIRECT_TO_ASH")
    });

    it("should render thank you redirect component when enable thank you redirect is true and the job has dsp enabled", () => {
        getFeatureFlagValueMock.mockReturnValueOnce(true);
        window.location.hash = `#/${PAGE_ROUTES.THANK_YOU}`;
        state = {
            ...initStoreState,
            application: TEST_APPLICATION_STATE,
            job: {
                ...TEST_JOB_STATE,
                results:{
                    ...TEST_JOB,
                    dspEnabled:true,
                }
            },
            candidate: TEST_CANDIDATE_STATE
        };
        const wrapper = mountWithStencil(
            <Provider store={store}>
                <ThankYouWithStore />
            </Provider>);

        expect(wrapper.find(ThankYouRedirectToAsh)).toHaveLength(1);
        expect(getFeatureFlagValueMock).toHaveBeenCalledWith("ENABLE_THANK_YOU_REDIRECT_TO_ASH")
    });
});