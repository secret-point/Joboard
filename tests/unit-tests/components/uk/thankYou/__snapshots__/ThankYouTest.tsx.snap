// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYouWrapper Component should match snapshot 1`] = `<Connect(ThankYouSummary) />`;
